src/js/core.js

```diff
@@ -981,6 +981,14 @@
                                     editableElementIndex: exportedSelection.editableElementIndex
                                 }
                             );
+                            // If textNodes are not present, when changing link on images
+                            // ex: <a><img src="http://image.test.com"></a>, change fragment to currRange.startContainer
+                            // and set textNodes array to [imageElement, imageElement]
+                            if (textNodes.length === 0) {
+                                fragment = this.options.ownerDocument.createDocumentFragment();
+                                fragment.appendChild(currRange.startContainer.cloneNode(true));
+                                textNodes = [fragment.firstChild.firstChild, fragment.firstChild.lastChild];
```

Comment:
Could `currRange.startContainer` ever be a text node?  If so, the reference to `fragment.firstChild.firstChild` and `fragment.firstChild.lastChild` could be undefined, in which case I'm not sure what the rest of the code would do.  It's also possible that this is handled ok, but I wanted to at least make sure you know what would happen in that case.